app/controllers/pieces_controller.rb

```diff
@@ -2,12 +2,17 @@ class PiecesController < ApplicationController
   before_action :set_game
   def update
     if selected_piece.move_to(params[:piece][:x_position].to_i,params[:piece][:y_position].to_i)
-      if selected_piece.game.has_started == false
-        selected_piece.game.has_started = true
-        selected_piece.game.save
-        Pusher["broadcast"].trigger!('game_has_started',{game_has_started: true})
-      end
-      
+        if selected_piece.can_pawn_promote?(params[:piece][:y_position].to_i)
+           Pusher["private-user_#{current_player.id}"].trigger!('pawn-promotion', {
+          pawnPromotionUpdateURL: game_piece_promote_pawn_path(@game, selected_piece.id), message: "You made it to the other side! Click one of the buttons to pick your new piece."
```

Comment:
I wonder if this is the line which is making it possible for me to promote white pawns in my current game, but not black pawns. When we're updating a piece, we want to notify the channel of whoever owns the piece, not necessarily the current_player.